### PR TITLE
Added oracle example

### PR DESCRIPTION
**Pull request by jackryanservia**

### CODE REVIEW

File: docs/zkapps/tutorials/oracle.mdx [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+Now that we are sure everything checks out, we can emit an event to indicate this. The first argument to `emitEvent()` is an arbitrary string name chosen by the developer (because a smart contract could emit more than one type of event) and the second argument can be any value, as long as it matches the type defined for our event earlier. In this case, our event is `Field`, but it could be a more complicated type built on Fields, if the situation called for it. Emitted events are stored and available on archive nodes in the Mina network.
+
+```ts
+// Emit an event containing the verified users id
+this.emitEvent('verified', id);
+```
+
+## Testing it Out
+
+The zkApp CLI automatically generated a file called `CreditScoreOracle.test.ts` for us when we ran `zk file CreditScoreOracle`. Let’s add some tests. Open it and paste in the following:
```
Comment (garethtdavies) [inline docs/zkapps/tutorials/oracle.mdx:282]: This should be `zk file CreditScoreVerifier` and `CreditScoreVerifier.test.ts` to match the `zk file` command from earlier.

File: docs/zkapps/tutorials/oracle.mdx [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+## Oracle
+
+### How it works
+
+Let’s start by understanding how an oracle works.
+
+Mina smart contracts run off-chain and make it possible to prove that the expected computation was run on private data without revealing the data itself. But if the smart contract consumes data from a 3rd party source, we need a way to verify that this data is authentic–i.e. that it was provided by the expected source.
+
+In the future, Mina’s zkOracles will allow a zkApp to consume data trustlessly from any HTTPS data source. But for now, the oracle design described in this tutorial will typically be operated by the zkApp developer, it will fetch and sign the desired data, and then a zkApp can consume this data and will verify the signature to ensure that the data was provided by the expected source.
+
```
Comment (etekis) [inline docs/zkapps/tutorials/oracle.mdx:37]: This also allows for data providers themselves to sign the data, without the zkApp developer having to fetch and act as an intermediary. In this way, when a KYC or credit score provider starts to sign their API calls, zkApp developers can easily consume that data while ensuring it is from the expected source.

**Review by etekis — CHANGES_REQUESTED** [review]

**Review by jackryanservia — COMMENTED** [review]
Added a note box to explain that data providers can also operate response signers so that users don't have to trust an intermediary party.

**Review by jasongitmail — COMMENTED** [review]
This looks good, but we still should change the repo name to `oracle-example`.

**Review by jasongitmail — APPROVED** [review]
Approving and merging in the interest of time. But @jackryanservia please follow up with renaming the repo to `oracle-example`. Thanks!

**Pull request merged**